src/magicbox/QueriesProcessor.ts

```diff
@@ -0,0 +1,91 @@
+import { defaults } from 'underscore';
+
+export interface IQueriesProcessorOptions {
+  timeout: number;
+}
+
+export enum QueryProcessResultStatus {
+  Finished = 'finished',
```

Comment:
Does it need to be a string enum ? 

Can it use string Typeguard instead ? 

```
const QueryProcessResultStatus = 'finished'|'timeout'|'overriden'
``` 

Does it even need to be a string, and a simple enum would do ? 

```
export enum QueryProcessResultStatus {
Finished,
TimedOut,
Overriden
}
```

Reply:
Good point, I'll change it to a simple enum. I don't remember why this was a string enum in the first place.